### PR TITLE
Added a docker image for using composer on devbox

### PR DESCRIPTION
**Pull request by Michal-Chatlas1221**

### CODE REVIEW

**Comment by Michal-Chatlas1221** [comment]
Decided to run composer as root for now. If we were to use host user we would need to wrap the application container to use it as well

File: docker/devbox/README.md [hunk 1]
```diff
@@ -112,3 +112,11 @@ php-tests images:
  ```bash
 docker build -f Dockerfile-tests -t artifactory.wikia-inc.com/platform/php-wikia-tests:latest .
  ```
+ 
+### Using composer
+The composer is not installed in the application container, so in order to install/update additional dependencies 
+you're going to want to use a different image, stored in `composer` directory. Example usage:
+
+```bash
+docker build -f composer/Dockerfile -t c ../../../app/ && docker run -it c php composer.phar require some/stuff
```
Comment (harnash) [inline docker/devbox/README.md]: I would split it into two separate commands / sections:
1. For running composer only (it should use image pushed to artifactory so we do not need to build this image every time (with some tag: `latest` perhaps)
2. For re-building the above image if needed (some configuration change or base was changed and needs rebuild, etc.)

File: docker/devbox/composer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+RUN php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');" \
+    && php composer-setup.php \
+    && php -r "unlink('composer-setup.php');" \
+    && apt-get update \
+    && apt-get install -y git zip unzip\
+    && pecl install yaml-2.0.2 xdebug-2.6.0 \
+    && docker-php-ext-enable yaml xdebug
+
+ADD . /usr/wikia/slot1/current/src
```
Comment (harnash) [inline docker/devbox/composer/Dockerfile]: It is common to use `VOLUME` in this case so you can reuse the image (we do that for devbox images). This way you can build one image and push it to repository and use it with your current app code.

File: docker/devbox/composer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+CMD ["php composer.phar"]
```
Comment (harnash) [inline docker/devbox/composer/Dockerfile]: I would use`gosu` https://github.com/Wikia/mobile-wiki/blob/dev/Dockerfile.devbox#L22 as @Grunny suggested to not mess up directory/files permissions.

**Review by harnash — CHANGES_REQUESTED** [review]

File: docker/devbox/composer/wrapper.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+USER_ID=${LOCAL_USER_ID:-9001}
+
+echo "Starting with UID : $USER_ID"
+adduser --home /app --shell /bin/bash --disabled-login --uid $USER_ID --system docker_user
+export HOME=/app
+
+set -Eeo pipefail
+
+gosu docker_user git config --global url."https://${GITHUB_TOKEN}@github.com/".insteadOf ssh://git@github.com/
```
Comment (harnash) [inline docker/devbox/composer/wrapper.sh]: I believe we do not need that here

**Review by harnash — APPROVED** [review]
One small remark. Rest LGTM

File: docker/devbox/composer/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    && apt-get install -y git zip unzip\
+    && pecl install yaml-2.0.2 xdebug-2.6.0 \
+    && docker-php-ext-enable yaml xdebug \
+    && apt-get install gosu
+
+VOLUME /usr/wikia/slot1/current/src
+VOLUME /usr/wikia/slot1/current/config
+
+ENTRYPOINT ["docker/devbox/composer/wrapper.sh"]
+CMD ["php composer.phar"]
```
Comment (harnash) [inline docker/devbox/composer/Dockerfile]: Shouldn't it be now like this?
```suggestion
CMD ["php composer"]
```

**Review by harnash — CHANGES_REQUESTED** [review]

**Review by harnash — APPROVED** [review]

**Pull request merged**